new_platforms/src/optee_common.c

```diff
@@ -7,17 +7,361 @@
 #include <openenclave/bits/report.h>
 #include <openenclave/bits/result.h>
 
+#include <mbedtls/sha256.h>
+#include <mbedtls/x509_crt.h>
+
+#include <RiotStatus.h>
+#include <RiotEcc.h>
+#include <RiotCrypt.h>
+#include <TcpsId.h>
+
 #ifndef _In_
 #include "sal_unsup.h"
 #endif
 
+static unsigned char tcps_oid[] = {0x67, 0x81, 0x05, 0x05, 0x04, 0x02};
+
+// based on mbedtls x509_get_crt_ext
```

Comment:
Had to put it back, there are two versions of mbedtls in the OE repo, one under 3rdparty and other under optee_os. As such, I do not want to maintain two copies of this code. 

So back to original question, is is ok here from a legal perspective? 

In parallel, I have opened a thread on mbedtls forum asking if they would want a function like that.